src/main/java/org/bytedeco/javacpp/Loader.java

```diff
@@ -1247,7 +1251,7 @@ public static String load(Class cls, Properties properties, boolean pathsFirst,
                     foundLibraries.put(preload, urls = findLibrary(cls, p, preload, pathsFirst));
                 }
                 String filename = null;
-                if (oldUrls == null || urls.length > 0) {
+                if (oldUrls == null && urls.length > 0) {
```

Comment:
I'm not sure the or was intentional or not, but before it called the following loadLibrary method when
1. `oldURLs==null and urls.length>0` , so the library in question was not searched before and found now.
2. `oldURLs==null and urls.length==0`, e.g. the library in question was not searched before but not found.
2. `oldURLs!=null and urls.length>0`, e.g. the library in question was searched before but and found.

For me it does not make sense to attempt to load the library when it was not found (case 2) or was already loaded (case 3). Or did I oversee something?

Reply:
If a previously found library failed to load, we should try to keep finding other versions that might be able to load.